### PR TITLE
Fix/identity service model validation

### PR DESCRIPTION
**Pull request by MGibson1**
## Type of change
- [x] Bug fix
- [ ] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other

## Objective
Asana Task: https://app.asana.com/0/1200804338582616/1201754397034782/f

We moved `register` and `prelogin` endpoints out of API and into Indentity with #1807, but failed to mirror that change in the bitwarden/web project. However, we did update our internal deployment functions to point `api/prelogin` and `api/register` to the identity project. This resulted in different MVC options for dev vs QA/prod deployments.

This PR adds our `ModelState.IsValid` filter to Identity as well as API. Identity has no public interface so that logic remains in API.

## Code changes
* **api/startup & Identity/\*\*/accountsController**: Just some comment and format cleaning
* **api/ModelStateFilterAttribute**: Extract common code to base class and add template method override for handling invalid ModelStates differently in API
* **identity/startup**: Add in model state filter
* **SharedWeb/ModelStateFilterAttribute**: Extract common code this this base class. Default invalid behavior is throwing internalAPI error

## Testing requirements
Retest invalid model attributes (spaces in email registration, too long, missing data, etc.) Should fail as before.

## Before you submit
- [x] I have checked for formatting errors (`dotnet tool run dotnet-format --check`) (required)
- [ ] If making database changes - I have also updated Entity Framework queries and/or migrations
- [ ] I have added **unit tests** where it makes sense to do so (encouraged but not required)
- [ ] This change requires a **documentation update** (notify the documentation team)
- [ ] This change has particular **deployment requirements** (notify the DevOps team)

# RC
Note: This PR will be pulled into `rc` to fix a regression defect

### CODE REVIEW

File: src/Api/Startup.cs [hunk 1]
```diff
@@ -124,7 +124,7 @@ public void ConfigureServices(IServiceCollection services)
             services.AddCoreLocalizationServices();
 
 #if OSS
-                services.AddOosServices();
+            services.AddOosServices();
```
Comment (justindbaur) [inline src/Api/Startup.cs:127]: I've noticed this a few times. 👍

File: src/Api/Utilities/ModelStateValidationFilterAttribute.cs [hunk 2]
```diff
@@ -6,33 +6,24 @@
 
 namespace Bit.Api.Utilities
 {
-    public class ModelStateValidationFilterAttribute : ActionFilterAttribute
+    public class ModelStateValidationFilterAttribute : SharedWeb.Utilities.ModelStateValidationFilterAttribute
     {
         private readonly bool _publicApi;
 
-        public ModelStateValidationFilterAttribute(bool publicApi)
+        public ModelStateValidationFilterAttribute(bool publicApi) : base()
```
Comment (justindbaur) [inline src/Api/Utilities/ModelStateValidationFilterAttribute.cs:13]: Is the `: base()` required?

**Review by justindbaur — CHANGES_REQUESTED** [review]
One small question.

File: src/Api/Utilities/ModelStateValidationFilterAttribute.cs (hunk 2 above)
Comment (MGibson1) [reply src/Api/Utilities/ModelStateValidationFilterAttribute.cs:13]: No it's not

**Review by justindbaur — APPROVED** [review]

**Pull request merged**